types/node/index.d.ts

```diff
@@ -2354,7 +2354,7 @@ declare module "net" {
         unref(): void;
         ref(): void;
 
-        remoteAddress: string;
```

Comment:
I think so too. 
In my code where I found Socket.remoteAddress is undefined, remotePort and remoteFamily are undefined when remoteAddress is undefined.